src/mlpack/methods/ne/neat.hpp

```diff
@@ -0,0 +1,974 @@
+        if (idxInChild == -1) {
+          childGenome.AddHiddenNeuron(momGenome.aNeuronGenes[idxInParent]);
+        }
+////printf("crossover 4\n");
+        // Add to neuron
+        idxInChild = childGenome.GetNeuronIndex(momGenome.aLinkGenes[i].ToNeuronId());
+        idxInParent = momGenome.GetNeuronIndex(momGenome.aLinkGenes[i].ToNeuronId());
+        if (idxInChild == -1) {
+          childGenome.AddHiddenNeuron(momGenome.aNeuronGenes[idxInParent]);
+        }
+////printf("crossover 5\n");
+        continue;
+      }
+
+      if (linkContainedInDad && randNum < 0.5) {
+        childGenome.AddLink(momGenome.aLinkGenes[i]);
+////printf("crossover 6\n");
+        // Add from neuron
+        ssize_t idxInChild = childGenome.GetNeuronIndex(momGenome.aLinkGenes[i].FromNeuronId());
+        ssize_t idxInParent = momGenome.GetNeuronIndex(momGenome.aLinkGenes[i].FromNeuronId());
+        if (idxInChild == -1) {
+          childGenome.AddHiddenNeuron(momGenome.aNeuronGenes[idxInParent]);
+        }
+////printf("crossover 7\n");
+        // Add to neuron
+        idxInChild = childGenome.GetNeuronIndex(momGenome.aLinkGenes[i].ToNeuronId());
+        idxInParent = momGenome.GetNeuronIndex(momGenome.aLinkGenes[i].ToNeuronId());
+        if (idxInChild == -1) {
+          childGenome.AddHiddenNeuron(momGenome.aNeuronGenes[idxInParent]);
+        }
+////printf("crossover 8\n");
+        continue;
+      }
+////printf("crossover 9\n");
+      if (linkContainedInDad && randNum >= 0.5) {
+        childGenome.AddLink(dadGenome.aLinkGenes[idx]);
+////printf("crossover 10\n");
+        // Add from neuron   TODO: make it a function?? check whether crossover is correct.
+        ssize_t idxInChild = childGenome.GetNeuronIndex(dadGenome.aLinkGenes[idx].FromNeuronId());
+        ssize_t idxInParent = dadGenome.GetNeuronIndex(dadGenome.aLinkGenes[idx].FromNeuronId());
+        if (idxInChild == -1) {
+          childGenome.AddHiddenNeuron(dadGenome.aNeuronGenes[idxInParent]);
+        }
+////printf("crossover 11\n");
+        // Add to neuron
+        idxInChild = childGenome.GetNeuronIndex(dadGenome.aLinkGenes[idx].ToNeuronId());
+        idxInParent = dadGenome.GetNeuronIndex(dadGenome.aLinkGenes[idx].ToNeuronId());
+        if (idxInChild == -1) {
+          childGenome.AddHiddenNeuron(dadGenome.aNeuronGenes[idxInParent]);
+        }
+////printf("crossover 12\n");
+        continue;
+      }  
+    }
+  }
+
+  // Crossover two genome to get one genome.
+  void Crossover(Genome& genome1, Genome& genome2, Genome& childGenome) {
+    if (CompareGenome(genome1, genome2)) {  // genome1 is better
+      CrossoverLinkAndNeuron(genome1, genome2, childGenome);
+    } else {
+      CrossoverLinkAndNeuron(genome2, genome1, childGenome);
+    }
+  }
+
+  // Measure two genomes' disjoint (including exceed).
+  // NOTICE: we can separate into disjoint and exceed. But currently maybe it is enough.
+  double Disjoint(Genome& genome1, Genome& genome2) {
```

Comment:
I plan to change this after NEAT passed Cart Pole problem. As I saw multiple implementations that the parameters for Exceed and Disjoint are the same. Thus, we can keep the current function for now and it won't influence the performance. After that, it can easily be changed.